specs/deposits.md

```diff
@@ -151,6 +151,13 @@ account will be increased, but no other state transition will occur.
 If `isSystemTransaction` in the deposit is set to `true`, the gas used by the deposit is unmetered.
 It must not be subtracted from the gas pool and the `usedGas` field of the receipt must be set to 0.
 
+Note for application developers: because `CALLER` and `ORIGIN` are set to `from`, the
```

Comment:
I think this is really important to note somewhere. I doubt it has any practical security implications because, as you note, the caller cannot execute any code. But I don't know if people are using this check for other stuff.